Source/Core/InputCommon/DynamicInputTextureConfiguration.cpp

```diff
@@ -0,0 +1,371 @@
+// Copyright 2019 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "InputCommon/DynamicInputTextureConfiguration.h"
+
+#include <optional>
+#include <sstream>
+#include <string>
+
+#include <fmt/format.h>
+#include <picojson.h>
+
+#include "Common/CommonPaths.h"
+#include "Common/File.h"
+#include "Common/FileUtil.h"
+#include "Common/Logging/Log.h"
+#include "Common/StringUtil.h"
+#include "Core/ConfigManager.h"
+#include "Core/Core.h"
+#include "InputCommon/ControllerEmu/ControllerEmu.h"
+#include "InputCommon/ImageOperations.h"
+#include "VideoCommon/RenderBase.h"
+
+namespace
+{
+std::string GetStreamAsString(std::ifstream& stream)
+{
+  std::stringstream ss;
+  ss << stream.rdbuf();
+  return ss.str();
+}
+
+std::optional<int> GetColorCodeFromString(const std::string& str)
+{
+  if (!str.empty() && str[0] == '#')
+  {
+    const std::string color_code = str.substr(1);
+    if (color_code.size() == 6)
+    {
+      return std::stoi(color_code, 0, 16);
+    }
+    else if (color_code.size() == 8)
```

Comment:
```suggestion
    if (color_code.size() == 8)
```

else isn't necessary after a return, and our convention is to get rid of those returns

Reply:
Thanks.  That's a good point but I'm going to keep it because if I use `TryParse` there's the chance that the parse will fail and if that happens I don't want to take the next branch.